File: shopping-interface/src/main/java/com/shopping/core/service/BrandService.java
package com.shopping.core.service;

import com.shopping.core.pojo.good.Brand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    //查询品牌所有
    public List<Brand> findAll();

    PageResult findPage(Integer pageNum, Integer pageSize);

    void save(Brand brand);

    Brand findOne(Long id);

    void update(Brand brand);

    void delByIds(Long[] ids);

    PageResult search(Integer pageNum, Integer pageSize, Brand brand);

    List<Map> findList();
}

File: shopping-web-manager/src/main/webapp/js/service/BrandService.js
app.service("brandService", function ($http) {

    this.search = function (currentPage, itemsPerPage, searchEntity) {
        return $http.post("/brand/search.do?pageNum=" + currentPage + "&pageSize=" +
            itemsPerPage, searchEntity);
    }

    this.save = function (entity) {
        return $http.post("/brand/save.do", entity);
    }

    this.update = function (entity) {
        return $http.post("/brand/update.do", entity)
    }

    this.findOne = function (id) {
        return $http.get("/brand/findOne.do?id=" + id);
    }

    this.delIds = function (selectedIds) {
        return $http.get("/brand/deleteIds.do?ids=" + selectedIds)
    }
    this.selectOptionList =function () {
        return $http.get("/brand/findList.do")
    }
})